### PR TITLE
Pocket Pouches

### PR DESCRIPTION
**Pull request by Geevies**
* Ported pouches from CM, these are storage items that fit in your pockets. A bunch have been mapped into the cargo warehouse.

This screenshot is mostly accurate, the items on the top right rack have changed however.
image

### Asset Licenses
The following assets that **have not** been created by myself are included in this PR:

| Path | Original Author | License |
| --- | --- | --- |
| icons/obj/clothing/pouches.dmi | CM-SS13 | [Creative Commons 3.0 BY-SA](https://creativecommons.org/licenses/by-sa/3.0/) |

### CODE REVIEW

**Comment by Geevies** [comment]
!review

**Comment by DreamySkrell** [comment]
image

The pouches are cool, but as is I don't really see the point/purpose of these honestly:

- If these are supposed to be standard equipment, then they should spawn in crew's lockers, equipment vendors, maybe give them these on spawn. Otherwise, why do they spawn in the warehouse? Are people supposed to go to the warehouse every shift to get these?

- If these are not supposed to be standard equipment, then they should be orderable from cargo instead, and not be put in the warehouse.

**Comment by DreamySkrell** [comment]
image
This file is huge. Code for the pouches should be put in their own folder, and with different pouches types in their own files

File: code/game/objects/items/weapons/storage/pouch.dm [hunk 1]
```diff
@@ -0,0 +1,454 @@
+/obj/item/storage/pouch
+	name = "abstract pouch"
+	desc = "The physical manifestation of a concept of a pouch. Woah."
+	icon = 'icons/obj/clothing/pouches.dmi'
+	icon_state = "small_drop"
+	w_class = ITEMSIZE_LARGE //does not fit in backpack
+	max_w_class = ITEMSIZE_SMALL
+	slot_flags = SLOT_POCKET
+	storage_slots = 1
```
Comment (DreamySkrell) [inline code/game/objects/items/weapons/storage/pouch.dm:9]: I feel this is really "gamey" and makes no sense. 

Pouches hold a few small items, and you can put them in your pocket... but you can't put them in your backpack? So how big are they exactly? Belts are bigger than these surely, but can be put in your backpack.

**Comment by DreamySkrell** [comment]
image

I do not think these fit stylistically and palette-wise on Horizon. We do not use these colors like, anywhere. 
At the very least these should be recolored (not necessarily resprited) to corporate colors (blue-ish/gray-ish). There could be green variants for other factions, but I think these do not fit on Horizon to be crew equipment.

**Review by DreamySkrell — CHANGES_REQUESTED** [review]
as above

File: code/modules/mob/inventory.dm [hunk 2]
```diff
@@ -371,6 +371,9 @@ var/list/slot_equipment_priority = list( \
 	if(!item)
 		return FALSE
 
+	if(SEND_SIGNAL(src, COMSIG_MOVABLE_PRE_THROW, item) & COMPONENT_CANCEL_THROW)
+		return
```
Comment (FluffyGhoster) [inline code/modules/mob/inventory.dm:375]: This signal goes in /atom/movable/proc/throw_at

File: code/game/objects/items/weapons/storage/pouch.dm [hunk 3]
```diff
@@ -0,0 +1,454 @@
+/obj/item/storage/pouch
+	name = "abstract pouch"
```
Comment (FluffyGhoster) [inline code/game/objects/items/weapons/storage/pouch.dm:2]: If this is an abstract, it needs to be marked as such with `abstract_type`

File: code/game/objects/items/weapons/storage/pouch.dm [hunk 4]
```diff
@@ -0,0 +1,454 @@
...
+
+	/// The X offset of the gun overlay
+	var/icon_x = 0
+
+	/// The Y offset of the gun overlay
+	var/icon_y = -3
+
+/obj/item/storage/pouch/pistol/Destroy()
+	gun_underlay = null
+	current_gun = null
```
Comment (FluffyGhoster) [inline code/game/objects/items/weapons/storage/pouch.dm:123]: you need to either QDEL_NULL it or move it out of the pouch if the pouch is deleted, with a strong preference to the former

File: code/game/atoms.dm [hunk 5]
```diff
@@ -767,3 +767,7 @@
 /atom/proc/clear_bulletholes()
 	for(var/obj/effect/overlay/bmark/bullet_mark in src)
 		qdel(bullet_mark)
+
+/// things that object need to do when a movable atom inside it is deleted
+/atom/proc/on_stored_atom_del(atom/movable/AM)
```
Comment (FluffyGhoster) [inline code/game/atoms.dm:772]: Meaningfully name the proc parameter(s)

File: code/game/objects/items/weapons/storage/pouch.dm [hunk 6]
```diff
@@ -0,0 +1,454 @@
...
+	if(!.)
+		return
+	if(isgun(item))
+		var/obj/item/gun/pistol = item
+		if(!(pistol.slot_flags & SLOT_HOLSTER))
+			to_chat(user, SPAN_WARNING("\The [pistol] doesn't fit in \the [src]!"))
+			return FALSE
+		if(current_gun)
+			if(!stop_messages)
+				to_chat(usr, SPAN_WARNING("[src] already holds a gun."))
```
Comment (FluffyGhoster) [inline code/game/objects/items/weapons/storage/pouch.dm:142]:
```suggestion
				to_chat(user, SPAN_WARNING("[src] already holds a gun."))
```

Do not rely on the usr var, especially when you can pass in the correct user and are already doing that

**Review by FluffyGhoster — CHANGES_REQUESTED** [review]

**Review by SleepyGemmy — CHANGES_REQUESTED** [review]
seconding Dreamy. balance and effects on gameplay should also be considered. other than that, i'm not sure why this is needed when we already have many ways to expand your inventory.

**Comment by NonQueueingMatt** [comment]
A few things:
- In concept this is fine but storage absolutely needs to be nerfed as a whole if you're adding pocket pouches.
- It needs new sprites, because CM's are totally unfitting. Sammy can probably help you with that.

**Comment by Eyeveri** [comment]
Just going to throw this in too - instead of mapping them onto racks and such, please code them into a locker or a crate. 

I also agree with Dreamy's comments.

**Comment by Geevies** [comment]
this one will need a little more effort and love than i'm capable of giving it right now, but if someone wants to take it over with the good feedback provided above, feel free

**Pull request closed without merging**